src/Teapot.Web/Controllers/StatusController.cs

```diff
@@ -1,86 +1,55 @@
-            var allowedMethod = Request.Headers.Get("Access-Control-Request-Method") ?? "GET";
-            var allowedHeaders = Request.Headers.Get("Access-Control-Request-Headers") ?? "X-Anything";
-
-            var responseHeaders = new Dictionary<string, string>
-            {
-                { "Access-Control-Allow-Origin", allowedOrigin },
-                { "Access-Control-Allow-Headers", allowedHeaders },
-                { "Access-Control-Allow-Methods", allowedMethod }
-            };
-
-            var statusData = new StatusCodeResult { IncludeHeaders = responseHeaders };
-
-            DoSleep(sleep);
-
-            return new CustomHttpStatusCodeResult((int)HttpStatusCode.OK, statusData);
-        }
-
-        private static void DoSleep(int? sleep)
-        {
-            int sleepData = SanitizeSleepParameter(sleep, SLEEP_MIN, SLEEP_MAX);
-
-            if (sleepData > 0)
-            {
-                System.Threading.Thread.Sleep(sleepData);
-            }
-        }
-
-        private static int SanitizeSleepParameter(int? sleep, int min, int max)
-        {
-            var sleepData = sleep ?? 0;
-
-            // range check - minimum should be 0
-            if (sleepData < min)
-            {
-                sleepData = min;
-            }
-
-            // range check- maximum should be 300000 (5 mins)
-            if (sleepData > max)
-            {
-                sleepData = max;
-            }
-
-            return sleepData;
-        }
-    }
-}
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+using Teapot.Web;
+using Teapot.Web.Models;
+
+namespace Teapot.Web.Controllers
+{
+    //[EnableCors]
+    public class StatusController : Controller
+    {
+        private static readonly TeapotStatusCodeResults StatusCodes = new TeapotStatusCodeResults();
+
+        [HttpGet("")]
+        public IActionResult Index() => View(StatusCodes);
+
+        [Route("{statusCode}", Name = "StatusCode")]
+        public async Task<IActionResult> StatusCode(int statusCode, int? sleep)
+        {
+            var statusData = StatusCodes.ContainsKey(statusCode)
+                ? StatusCodes[statusCode]
+                : new TeapotStatusCodeResult { Description = $"{statusCode} Unknown Code" };
+
+            await DoSleep(sleep);
+
+            return new CustomHttpStatusCodeResult(statusCode, statusData);
+        }
+
+        [EnableCors(Constants.CorsPolicy)]
+        [Route("{statusCode}/cors", Name = "Cors")]
+        public async Task<IActionResult> Cors(int statusCode, int? sleep)
+        {
+            var statusData = StatusCodes.ContainsKey(statusCode)
+                ? StatusCodes[statusCode]
+                : new TeapotStatusCodeResult { Description = $"{statusCode} Unknown Code" };
+
+            await DoSleep(sleep);
+
+            return new CustomHttpStatusCodeResult(statusCode, statusData);
```

Comment:
We were also always returning a `200` back, regardless of the one in the route.

If I lack context and it was on purpose, let me know.

Reply:
For the CORS pre-flight (the `OPTIONS` request), it should always return a 200 response, as it's really the next-step that you want to test failure on.

Reply:
The preflight request will be entirely handled by the middleware, and won't reach MVC, which is why I switched to returning the provided status code.

Reply:
Maybe the right question is, what were we trying to achieve with that endpoint?

Reply:
The point of this endpoint was that we weren't handling CORS on the normal `/200` call, only if you hit `/200/cors`. If CORS is being lit up across the whole app (a better idea IMO), then we probably don't need it

Reply:
Got it, I've pushed an update which enable CORS at the app level.